src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -0,0 +1,260 @@
+      this.addCourseToSemester(season, year, newCourse);
+      this.$emit('addCourse', newCourse);
+    },
+    openCourseModal() {
+      this.isCourseModalOpen = true;
+    },
+    closeCourseModal() {
+      this.isCourseModalOpen = false;
+    },
+    checkCourseDuplicate(key: string) {
+      // if (this.courses) {
+      //   // @ts-ignore
+      //   this.$refs.modal.courseIsAddable = true;
+      //   this.courses.forEach(course => {
+      //     if (course.code === key) {
+      //       // @ts-ignore
+      //       this.$refs.modal.courseIsAddable = false;
+      //       this.$emit('open-caution-modal');
+      //     }
+      //   });
+      // }
+    },
+
+    addCourseToSemester(
+      season: FirestoreSemesterType,
+      year: number,
+      newCourse: FirestoreSemesterCourse
+    ) {
+      editSemesters(oldSemesters => {
+        let semesterFound = false;
+        const newSemestersWithCourse = oldSemesters.map(sem => {
+          if (sem.type === season && sem.year === year) {
+            semesterFound = true;
+            return { ...sem, courses: [...sem.courses, newCourse] };
+          }
+          return sem;
+        });
+        if (semesterFound) return newSemestersWithCourse;
+        return [...oldSemesters, this.createSemester([newCourse], season, year)].sort(this.compare);
+      });
+    },
+    createSemester(
+      courses: readonly FirestoreSemesterCourse[],
+      type: FirestoreSemesterType,
+      year: number
+    ) {
+      return { courses, type, year };
+    },
+    compare(a: FirestoreSemester, b: FirestoreSemester): number {
+      if (a.type === b.type && a.year === b.year) {
+        return 0;
+      }
+      if (a.year > b.year) {
+        return -1;
+      }
+      if (a.year < b.year) {
+        return 1;
+      }
+      if (SeasonsEnum[a.type] < SeasonsEnum[b.type]) {
+        return 1;
+      }
+      return -1;
+    },
+  },
+});
+</script>
+
+<style scoped lang="scss">
+@import '@/assets/scss/_variables.scss';
+
+.dropdown {
+  &-select {
+    display: flex;
+    flex-direction: row;
+    background: $white;
+    border: 0.5px solid $inactiveGray;
+    box-sizing: border-box;
+    border-radius: 2px;
+    width: 100%;
+    font-size: 14px;
+    line-height: 17px;
+    color: $darkPlaceholderGray;
+    position: relative;
+
+    &:not(:first-child) {
+      margin-top: 0.5rem;
+    }
+  }
+  &-placeholder {
+    height: 100%;
+    font-size: 14px;
+    line-height: 17px;
+    margin-left: 0.25rem;
+    display: flex;
+    align-items: center;
+    color: $darkPlaceholderGray;
+    background: transparent;
+    cursor: pointer;
+  }
+  &-wrapper {
+    display: flex;
+    flex-direction: row;
+    width: 100%;
+    height: 100%;
+  }
+  &-arrow {
+    border-left: 6.24px solid transparent;
+    border-right: 6.24px solid transparent;
+    border-top: 6.24px solid $inactiveGray;
+    background: transparent;
+    margin-right: 8.7px;
+    margin-left: 5px;
+    margin-top: 5px;
```

Comment:
Can we have margins in rem?

Reply:
It looks like these exact pixel numbers are used within 4 different dropdowns currently. I agree they should be in rem, but maybe we should update them if/when we factor out dropdown?